src/snowflake/snowpark/stored_procedure_profiler.py

```diff
@@ -89,7 +89,7 @@ def set_active_profiler(
         with self._lock:
             if self._query_history is None:
                 self._query_history = self._session.query_history(
-                    include_thread_id=True
```

Comment:
something I noticed now is that we are acquiring lock twice in this function. If it does not impact the functionality, can we do all this under one lock?